#!/usr/bin/env python
PACKAGE = "dynamic_tutorials"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("arm_inertia",    double_t,    0, "Joint inertia at arm side", 10,  0, 20)
gen.add("motor_inertia", double_t, 0, "Joint inertia at motor side",   10, 0,  20)
gen.add("arm_damp",    double_t,    0, "Damping coefficient at arm side",  5, 0, 10)
gen.add("motor_damp",    double_t,    0, "Damping coefficient at motor side",  5, 0, 10)
gen.add("arm_elastic",    double_t,    0, "Elastic coefficient at arm side",  2, 0, 10)
gen.add("motor_elastic",    double_t,    0, "Elastic coefficient at motor side",  2, 0, 10)
gen.add("friction_torque",    double_t,    0, "Friction torque",  1, 0, 10)
gen.add("gear_ratio",    int_t,    0, "Gear box reduction ratio",  20, 0, 100)

exit(gen.generate(PACKAGE, "dynamic_paras", "Tutorials"))
